tsc fails on fresh unstable-vite install due to remix & vite typescript conflicts

### Reproduction

Compiling a fresh install of unstable-vite remix as follows:
```
npx create-remix@latest --template remix-run/remix/templates/unstable-vite vitetest
cd vitetest
# Using Typescript (tsc) 5.3.2
npx tsc
```

results in 49 typescript errors due to module definition conflicts between remix and vite:
```
Found 49 errors in 4 files.

Errors  Files
    23  node_modules/@remix-run/dev/dist/modules.d.ts:3
     2  node_modules/@remix-run/react/dist/browser.d.ts:30
    23  node_modules/vite/client.d.ts:8
     1  node_modules/vite/types/importMeta.d.ts:17
```


### System Info

```shell
NOTE: Tested with Typescript 5.3.2

System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M2
    Memory: 4.35 GB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.3.1 - ~/.nvm/versions/node/v20.3.1/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v20.3.1/bin/yarn
    npm: 9.6.7 - ~/.nvm/versions/node/v20.3.1/bin/npm
  Browsers:
    Chrome: 120.0.6099.129
    Safari: 17.2.1
  npmPackages:
    @remix-run/dev: ^2.4.1 => 2.4.1
    @remix-run/node: ^2.4.1 => 2.4.1
    @remix-run/react: ^2.4.1 => 2.4.1
    @remix-run/serve: ^2.4.1 => 2.4.1
    vite: ^5.0.0 => 5.0.11
```


### Used Package Manager

npm

### Expected Behavior

Compiling a fresh install of unstable-vite with `npx tsc` should complete successfully.
```
npx create-remix@latest --template remix-run/remix/templates/unstable-vite vitetest
cd vitetest
# NOTE: Using Typescript (tsc) 5.3.2
npx tsc
```

### Actual Behavior

```
Found 49 errors in 4 files.

Errors  Files
    23  node_modules/@remix-run/dev/dist/modules.d.ts:3
     2  node_modules/@remix-run/react/dist/browser.d.ts:30
    23  node_modules/vite/client.d.ts:8
     1  node_modules/vite/types/importMeta.d.ts:17
```


There are basically just 2 errors: 1 instance of the following type declaration mismatch:
```
node_modules/@remix-run/react/dist/browser.d.ts:30:9 - error TS2687: All declarations of 'hot' must have identical modifiers.
node_modules/@remix-run/react/dist/browser.d.ts:30:9 - error TS2717: Subsequent property declarations must have the same type.  Property 'hot' must be of type 'ViteHotContext | undefined', but here has type 'any'.

30         hot: any;
           ~~~
```

and 23 instances of duplicate module definitions:
```
node_modules/@remix-run/dev/dist/modules.d.ts:3:20 - error TS2300: Duplicate identifier 'asset'.

3     export default asset;
                     ~~~~~

  node_modules/vite/client.d.ts:129:18
    129   export default src
                         ~~~

```

--- a/docs/future/vite.md
+++ b/docs/future/vite.md
@@ -134,6 +134,8 @@ export default defineConfig({
 
 Vite handles imports for all sorts of different file types, sometimes in ways that differ from the existing Remix compiler, so let's reference Vite's types from `vite/client` instead of the obsolete types from `@remix-run/dev`.
 
+Since the module types provided by `vite/client` are not compatible with the module types implicitly included with `@remix-run/dev`, you'll also need to enable the `skipLibCheck` flag in your TypeScript config. Remix won't require this flag in the future once the Vite plugin is the default compiler.
+
 👉 **Rename `remix.env.d.ts` to `env.d.ts`**
 
 ```diff nonumber
@@ -154,6 +156,12 @@ Vite handles imports for all sorts of different file types, sometimes in ways th
 ```diff filename=tsconfig.json
 - "include": ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
 + "include": ["env.d.ts", "**/*.ts", "**/*.tsx"],
+```
+
+👉 **Ensure `skipLibCheck` is enabled in `tsconfig.json`**
+
+```json filename=tsconfig.json
+"skipLibCheck": true,
 ```
 
 👉 **Ensure `module` and `moduleResolution` fields are set correctly in `tsconfig.json`**

--- a/templates/unstable-vite-express/tsconfig.json
+++ b/templates/unstable-vite-express/tsconfig.json
@@ -11,20 +11,12 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {
       "~/*": ["./app/*"]
     },
-
-    // The `skipLibCheck` flag is enabled in Remix's Vite templates due to a
-    // clash between the module types from `vite/client` and the old module
-    // types from `@remix-run/dev` which were intended for use with Remix's
-    // esbuild compiler. When `@remix-run/dev` no longer implicitly includes
-    // these module types, the Remix Vite templates will no longer enable this
-    // flag. This will most likely happen in a future major release when the
-    // esbuild compiler is removed in favor of the Vite plugin.
-    "skipLibCheck": true,
 
     // Vite takes care of building everything, not tsc.
     "noEmit": true

--- a/templates/unstable-vite-express/tsconfig.json
+++ b/templates/unstable-vite-express/tsconfig.json
@@ -17,7 +17,16 @@
       "~/*": ["./app/*"]
     },
 
-    // Remix takes care of building everything in `remix build`.
+    // The `skipLibCheck` flag is enabled in Remix's Vite templates due to a
+    // clash between the module types from `vite/client` and the old module
+    // types from `@remix-run/dev` which were intended for use with Remix's
+    // esbuild compiler. When `@remix-run/dev` no longer implicitly includes
+    // these module types, the Remix Vite templates will no longer enable this
+    // flag. This will most likely happen in a future major release when the
+    // esbuild compiler is removed in favor of the Vite plugin.
+    "skipLibCheck": true,
+
+    // Vite takes care of building everything, not tsc.
     "noEmit": true
   }
 }

--- a/templates/unstable-vite/tsconfig.json
+++ b/templates/unstable-vite/tsconfig.json
@@ -11,20 +11,12 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {
       "~/*": ["./app/*"]
     },
-
-    // The `skipLibCheck` flag is enabled in Remix's Vite templates due to a
-    // clash between the module types from `vite/client` and the old module
-    // types from `@remix-run/dev` which were intended for use with Remix's
-    // esbuild compiler. When `@remix-run/dev` no longer implicitly includes
-    // these module types, the Remix Vite templates will no longer enable this
-    // flag. This will most likely happen in a future major release when the
-    // esbuild compiler is removed in favor of the Vite plugin.
-    "skipLibCheck": true,
 
     // Vite takes care of building everything, not tsc.
     "noEmit": true

--- a/templates/unstable-vite/tsconfig.json
+++ b/templates/unstable-vite/tsconfig.json
@@ -17,7 +17,16 @@
       "~/*": ["./app/*"]
     },
 
-    // Remix takes care of building everything in `remix build`.
+    // The `skipLibCheck` flag is enabled in Remix's Vite templates due to a
+    // clash between the module types from `vite/client` and the old module
+    // types from `@remix-run/dev` which were intended for use with Remix's
+    // esbuild compiler. When `@remix-run/dev` no longer implicitly includes
+    // these module types, the Remix Vite templates will no longer enable this
+    // flag. This will most likely happen in a future major release when the
+    // esbuild compiler is removed in favor of the Vite plugin.
+    "skipLibCheck": true,
+
+    // Vite takes care of building everything, not tsc.
     "noEmit": true
   }
 }

